config/site/src/guides/datetime-best-practices.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: markdown
+title: DateTime Best Practices
+permalink: /guides/datetime-best-practices/
+nav_title: DateTime Best Practices
+menu_order: 25
+---
+
+This guide covers important considerations when working with DateTime fields in ElasticGraph, particularly when using `min_value` or `max_value` derived fields.
+
+## String-Based Comparison for Min/Max DateTime Values
+
+When you use [`derive_indexed_type_fields`](https://rubydoc.info/gems/elasticgraph-schema_definition/ElasticGraph/SchemaDefinition/Indexing/DerivedIndexedType) with `min_value` or `max_value` on a DateTime field, ElasticGraph uses string comparison via Painless's `.compareTo()` method to determine which value is smaller or larger.
+
+This works correctly for ISO-8601 formatted DateTime strings **as long as the precision is consistent**.
+
+### ElasticGraph Handles This Automatically
+
+ElasticGraph normalizes all DateTime values to fixed 3-digit millisecond precision at two levels:
+
+1. **Indexer path**: The [`DateTime` indexing preparer](https://github.com/block/elasticgraph/blob/main/elasticgraph-indexer/lib/elastic_graph/indexer/indexing_preparers/date_time.rb) normalizes publisher data before it's indexed.
+
+2. **GraphQL path**: The [`DateTime` scalar coercion adapter](https://github.com/block/elasticgraph/blob/main/elasticgraph-graphql/lib/elastic_graph/graphql/scalar_coercion_adapters/date_time.rb) normalizes query filter values.
+
+Both use 3-digit millisecond precision:
+
+```ruby
+PRECISION = 3 # millisecond precision
+time.iso8601(PRECISION)  # Always outputs 3-digit milliseconds
+```
+
+This ensures that:
+- Data from publishers is normalized before indexing
+- Query filter values passed through GraphQL are normalized
+- String comparisons work correctly for min/max tracking
+
+**If you're using ElasticGraph's standard indexer and GraphQL API, you don't need to worry about this issue.**
+
+### When You Need to Be Careful: Query Interceptors
```

Comment:
It seems like this is the only thing remaining that EG doesn't automatically handle for users. Is there a way we can improve EG to handle this case so that users don't have to be aware of it?